core/src/main/kotlin/org/evomaster/core/output/TestCaseWriter.kt

```diff
@@ -455,40 +469,35 @@ class TestCaseWriter {
                 when (bodyString?.first()) {
                     '[' -> {
                         // This would be run if the JSON contains an array of objects.
-                        // Only assertions on array size are supporte at the moment.
                         val resContents = Gson().fromJson(res.getBody(), ArrayList::class.java)
                         lines.add(".body(\"size()\", equalTo(${resContents.size}))")
-                        if (resContents.size > 0) {
-                            resContents.forEachIndexed { index, value ->
-                                val test_i = index
-                                val printableTh = handleFieldValues(value)
-                                if (printableTh != "null"
-                                        && printableTh != "NotCoveredYet"
-                                        && !printableTh.contains("logged")
-                                ) {
-                                    lines.add(".body(\"get($test_i)\", $printableTh)")
+                        //resContents.sortBy { it.toString() }
+                        //assertions on contents
+                        if(resContents.size > 0){
+                            if(resContents.first()::class == LinkedTreeMap::class) resContents.sortBy { it.toString() }
+                            // Sorting needed as sometimes retrieving collections results in non-deterministic order
+                            // (eg ScoutAPI - users).
```

Comment:
even if we sort it here when we generate the JUnit, how can guarantee to have same order when running the generated tests?

Reply:
At the moment, I am doing a basic lookup (as we discussed today). 

The assertion is that in the collection retrieved, there is one element that has a particular value. (e.g. for the field "id", the collection x.id contains one matching element). 

I will write up a more detailed comment on this, as the current approach could be improved. It should, however, work for basic assertions.

Reply:
after the discussion we had, shouldn't we remove this sorting here?